compiler-core/src/language_server/engine.rs

```diff
@@ -219,6 +219,64 @@ where
         })
     }
 
+    pub fn action(
+        &mut self,
+        params: lsp::CodeActionParams,
+    ) -> Response<Option<Vec<lsp_types::CodeAction>>> {
+        eprintln!("Got a new action request: {:?}", params);
+        self.respond(|this| {
+            let mut actions = vec![];
+
+            // question: how to get the current module warnings?
+            // These could be used to generate a single code action to fix all the unused warnings at once?
+            eprintln!("Why is this empty? {:?}", this.compiler.warnings);
```

Comment:
Warnings are events that are dispatched to the user, so this vector is an IO buffer that has recently been flushed.

The analyser will need to be modified to store the locations of all unused imports, and then that can be used here.

Reply:
Great, thank you for the explanation. So I did that in d0e9718, and the action now seems to work for multiple unused imported module. However it's a bit annoying because neither the ast nor the warning contains the "import" token, so I had to do some string lookups to make it works. If that looks good to you, then I'd like to add some tests and perhaps also handle unused imported values.

Reply:
Ok, like this https://github.com/gleam-lang/gleam/pull/2325 ?

Reply:
Perfect!